lua/nvim-surround/config.lua

```diff
@@ -345,6 +354,14 @@ M.get_selections = function(args)
     end
 end
 
+-- Translate Vim-style key notation to terminal code, for usage in `surrounds` keys.
```

Comment:
Do I also need to copy this to the docs, or do you have some tool that regenerates them?

Reply:
Hmm... I was thinking perhaps instead of this implementation we could *always* pass the input through `nvim_replace_termcodes`? To my knowledge, that should do nothing for normal characters, and set things properly for control characters. It would add a bit of "syntactic sugar" I think, since you would be able to bind surrounds directly using `<BS>` or `<CR>`.

Reply:
Passing the input through `nvim_replace_termcodes` is not good. We need to either pass the input through `keytrans`, or pass the output configuration through `nvim_replace_termcodes`. I''m against the former, because:

1. Using `keytrans` on the input will create casing pitfall - one could set `surrounds` for `<cr>` and `<bs>`, expecting them to work because they work for keybindings, but they'll fail because nvim-surround will look for `<CR>` and `<BS>`.
2. `surrounds` like `"<"` or `"\r"` will fail because the input will be translated to `<lt>` and `<CR>` (BTW: Note that `<lt>` is lowercase while `<CR>` is uppercase. That's a big inconsistency on `keytrans`' part which only emphasizes my first problem with it)

The latter is not without problems - processing a configuration tends to have its sharp corners - but I see that you are already doing it in `config.translate_opts`, so maybe it won't be that bad?

Reply:
@kylechui I've added a new commit to this PR where `config.termcode` is no longer required because the translation is done automatically in `config.translate_opts`.